tests/etl/test_save.py

```diff
@@ -10,162 +11,271 @@
 from unified_graphics.models import Analysis, WeatherModel
 
 
-@pytest.fixture
-def zarr_file(tmp_path):
-    return tmp_path / "unified_graphics.zarr"
-
-
 @pytest.fixture(scope="module")
-def analysis():
-    init_time = "2022-05-05T14:00"
-    model = "RTMA"
+def model():
+    mdl = "RTMA"
     system = "WCOSS"
     domain = "CONUS"
     frequency = "REALTIME"
     background = "HRRR"
 
-    return (init_time, model, system, domain, frequency, background)
+    return (mdl, system, domain, background, frequency)
 
 
-@pytest.fixture
-def ps_anl(analysis, diag_dataset, session, zarr_file):
-    (init_time, model, system, domain, frequency, background) = analysis
-    ps = diag_dataset(
-        "ps", init_time, "anl", model, system, domain, frequency, background
-    )
+@pytest.fixture(scope="class")
+def data_path(tmp_path_factory):
+    return tmp_path_factory.mktemp("test_etl_save")
 
-    diag.save(session, zarr_file, ps)
 
-    return ps
+@pytest.fixture(scope="class")
+def zarr_file(data_path):
+    return data_path / "unified_graphics.zarr"
 
 
-@pytest.fixture
-def ps_ges(analysis, diag_dataset, session, zarr_file):
-    (init_time, model, system, domain, frequency, background) = analysis
-    ps = diag_dataset(
-        "ps", init_time, "ges", model, system, domain, frequency, background
-    )
+@pytest.fixture(scope="class")
+def parquet_file(model, data_path):
+    (mdl, system, domain, background, frequency) = model
+    return data_path / "_".join((mdl, background, system, domain, frequency))
 
-    diag.save(session, zarr_file, ps)
 
-    return ps
+def dataset_to_table(dataset: xr.Dataset) -> pd.DataFrame:
+    df = dataset.to_dataframe()
+    df["loop"] = dataset.loop
 
+    return df.astype({"loop": "category"})
 
-@pytest.fixture
-def t_anl(analysis, diag_dataset, session, zarr_file):
-    (init_time, model, system, domain, frequency, background) = analysis
-    t = diag_dataset(
-        "t", init_time, "anl", model, system, domain, frequency, background
-    )
 
-    diag.save(session, zarr_file, t)
+class TestSaveNew:
+    @pytest.fixture(scope="class", autouse=True)
+    def dataset(self, model, diag_dataset, session, zarr_file):
+        (mdl, system, domain, background, frequency) = model
+        ps = diag_dataset(
+            "ps",
+            "2022-05-05T14:00",
+            "anl",
+            mdl,
+            system,
+            domain,
+            frequency,
+            background,
+        )
```

Comment:
This is a nice way to consolidate setup and handle multiple `assert`'s. Thanks for the doc link!